domains/noto/internal/noto/handler_delegate_lock_test.go

```diff
@@ -0,0 +1,499 @@
+	assert.JSONEq(t, expectedFunction, prepareRes.Transaction.FunctionAbiJson)
+	assert.Nil(t, prepareRes.Transaction.ContractAddress)
+
+	var paramsMap map[string]interface{}
+	err = json.Unmarshal([]byte(prepareRes.Transaction.ParamsJson), &paramsMap)
+	require.NoError(t, err)
+	assert.NotNil(t, paramsMap["unlockHash"]) // V0 includes unlockHash
+	assert.Equal(t, delegateAddress, paramsMap["delegate"])
+	assert.Equal(t, signatureBytes.String(), paramsMap["signature"])
+}
+
+// TestDelegateLockV1MintLock tests that delegating a mint lock (with no input states) fails at Assemble stage
+func TestDelegateLockV1MintLock(t *testing.T) {
+	n := &Noto{
+		Callbacks:        mockCallbacks,
+		coinSchema:       &prototk.StateSchema{Id: "coin"},
+		lockedCoinSchema: &prototk.StateSchema{Id: "lockedCoin"},
+		lockInfoSchemaV0: &prototk.StateSchema{Id: "lockInfo"},
+		lockInfoSchemaV1: &prototk.StateSchema{Id: "lockInfo_v1"},
+		dataSchemaV0:     &prototk.StateSchema{Id: "data"},
+		dataSchemaV1:     &prototk.StateSchema{Id: "data_v1"},
+	}
+	ctx := context.Background()
+	fn := types.NotoABI.Functions()["delegateLock"]
+
+	notaryAddress := "0x1000000000000000000000000000000000000000"
+	delegateAddress := "0x2000000000000000000000000000000000000000"
+	senderKey, err := secp256k1.GenerateSecp256k1KeyPair()
+	require.NoError(t, err)
+
+	lockID := pldtypes.RandBytes32()
+
+	// Mock returns no states - this is a mint lock with no input states
+	mockCallbacks.MockFindAvailableStates = func() (*prototk.FindAvailableStatesResponse, error) {
+		return &prototk.FindAvailableStatesResponse{
+			States: []*prototk.StoredState{},
+		}, nil
+	}
+
+	contractAddress := "0xf6a75f065db3cef95de7aa786eee1d0cb1aeafc3"
+	tx := &prototk.TransactionSpecification{
+		TransactionId: "0x015e1881f2ba769c22d05c841f06949ec6e1bd573f5e1e0328885494212f077d",
+		From:          "sender@node1",
+		ContractInfo: &prototk.ContractInfo{
+			ContractAddress: contractAddress,
+			ContractConfigJson: mustParseJSON(&types.NotoParsedConfig{
+				NotaryLookup: "notary@node1",
+				NotaryMode:   types.NotaryModeBasic.Enum(),
+				Variant:      types.NotoVariantDefault,
+			}),
+		},
+		FunctionAbiJson:   mustParseJSON(fn),
+		FunctionSignature: fn.SolString(),
+		FunctionParamsJson: fmt.Sprintf(`{
+			"lockId": "%s",
+			"delegate": "%s",
+			"data": "0x1234"
+		}`, lockID, delegateAddress),
+	}
+
+	verifiers := []*prototk.ResolvedVerifier{
+		{
+			Lookup:       "notary@node1",
+			Algorithm:    algorithms.ECDSA_SECP256K1,
+			VerifierType: verifiers.ETH_ADDRESS,
+			Verifier:     notaryAddress,
+		},
+		{
+			Lookup:       "sender@node1",
+			Algorithm:    algorithms.ECDSA_SECP256K1,
+			VerifierType: verifiers.ETH_ADDRESS,
+			Verifier:     senderKey.Address.String(),
+		},
+	}
+
+	assembleRes, err := n.AssembleTransaction(ctx, &prototk.AssembleTransactionRequest{
+		Transaction:       tx,
+		ResolvedVerifiers: verifiers,
+	})
+
+	// Currently expecting REVERT result (this is wrong!)
```

Comment:
Note that the test reverts here - but we need to find a way to allow this pattern (delegating a lock that has no input states).